Index: demoscene-toolchain/submodules/binutils-gdb/gdb/dbxread.c
===================================================================
--- demoscene-toolchain.orig/submodules/binutils-gdb/gdb/dbxread.c
+++ demoscene-toolchain/submodules/binutils-gdb/gdb/dbxread.c
@@ -2622,7 +2622,7 @@ process_one_symbol (int type, int desc,
       /* Relocate for dynamic loading and for ELF acc
 	 function-relative symbols.  */
 #if (DEFAULT_BFD_VEC != amiga_vec)
-      valu += function_start_offset;
+      /* valu += function_start_offset; */
 #endif
 
       /* GCC 2.95.3 emits the first N_SLINE stab somewhere in the
