--- demoscene-toolchain.orig/submodules/binutils-gdb/gdb/amigaread.c
+++ demoscene-toolchain/submodules/binutils-gdb/gdb/amigaread.c
@@ -374,7 +374,7 @@ amiga_symtab_read (minimal_symbol_reader &reader,
 	     symbols in their own section, so all symbols we are
 	     interested in will have a section.  */
 	  /* Bfd symbols are section relative.  */
-	  symaddr = sym->value + sym->section->vma;
+	  symaddr = sym->value;
 	  /* For non-absolute symbols, use the type of the section
 	     they are relative to, to intuit text/data.  Bfd provides
 	     no way of figuring this out for absolute symbols.  */
