Index: demoscene-toolchain/submodules/fs-uae/src/od-fs/uaemisc.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/od-fs/uaemisc.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/od-fs/uaemisc.cpp
@@ -10,6 +10,7 @@
 #include "xwin.h"
 #include "clipboard.h"
 #include "uae/fs.h"
+#include "fs/conf.h"
 #include "../od-win32/debug_win32.h"
 
 #include <fs/emu/hacks.h>
@@ -203,7 +204,6 @@ TCHAR *target_expand_environment (const
 	}
 }
 
-#if 0
 #include <signal.h>
 #include "debug.h"
 #ifdef __cplusplus_disabled
@@ -220,11 +220,12 @@ static RETSIGTYPE sigbrkhandler (int foo
     signal (SIGINT, sigbrkhandler);
 #endif
 }
-#endif
 
 void setup_brkhandler (void)
 {
-    /*
+    if (fs_config_false("console_debugger"))
+        return;
+
 #if defined(__unix) && !defined(__NeXT__)
     struct sigaction sa;
     sa.sa_handler = sigbrkhandler;
@@ -237,7 +238,6 @@ void setup_brkhandler (void)
 #else
     signal (SIGINT, sigbrkhandler);
 #endif
-    */
 }
 
 void sleep_cpu_wakeup(void)
