Index: demoscene-toolchain/submodules/fs-uae/src/fsuae/fsuae-main.c
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/fsuae/fsuae-main.c
+++ demoscene-toolchain/submodules/fs-uae/src/fsuae/fsuae-main.c
@@ -803,6 +803,10 @@ static void on_init(void)
     fs_uae_configure_input();
     fs_uae_configure_directories();
 
+    if (fs_config_true("use_debugger")) {
+        amiga_set_option("use_debugger", "yes");
+    }
+
     if (fs_config_get_int("save_state_compression") == 0) {
         amiga_set_save_state_compression(0);
     } else {
Index: demoscene-toolchain/submodules/fs-uae/src/od-fs/stubs.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/od-fs/stubs.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/od-fs/stubs.cpp
@@ -179,7 +179,7 @@ void target_fixup_options(struct uae_pre
 }
 
 int debuggable (void) {
-    return 0;
+    return 1;
 }
 
 void logging_init(void) {
