Index: demoscene-toolchain/submodules/fs-uae/src/od-fs/writelog_fs.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/od-fs/writelog_fs.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/od-fs/writelog_fs.cpp
@@ -479,7 +479,7 @@ void console_out_f (const TCHAR *format,
 #else
     va_list arg_ptr;
     va_start(arg_ptr, format);
-    vprintf(format, arg_ptr);
+    vfprintf(stderr, format, arg_ptr);
     va_end(arg_ptr);
 
 #endif
@@ -490,7 +490,7 @@ void console_out (const TCHAR *txt)
 #ifdef _WIN32
 	console_put (txt);
 #else
-	printf("%s", txt);
+	fprintf(stderr, "%s", txt);
 #endif
 }
 
@@ -596,7 +596,7 @@ void console_flush (void)
 #ifdef _WIN32
 	flushconsole ();
 #else
-	fflush(stdout);
+	fflush(stderr);
 #endif
 }
 
