Index: demoscene-toolchain/submodules/fs-uae/src/cpuboard.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/cpuboard.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/cpuboard.cpp
@@ -1055,12 +1055,16 @@ void cpuboard_rethink(void)
 			safe_interrupt_set(IRQ_SOURCE_CPUBOARD, 0, false);
 			if (currprefs.cachesize)
 				atomic_or(&uae_int_requested, 0x010000);
+#ifdef WITH_PPC
 			uae_ppc_wakeup_main();
+#endif
 		} else if (!(io_reg[CSIII_REG_IRQ] & (P5_IRQ_PPC_1 | P5_IRQ_PPC_2))) {
 			safe_interrupt_set(IRQ_SOURCE_CPUBOARD, 1, false);
 			if (currprefs.cachesize)
 				atomic_or(&uae_int_requested, 0x010000);
+#ifdef WITH_PPC
 			uae_ppc_wakeup_main();
+#endif
 		} else {
 			atomic_and(&uae_int_requested, ~0x010000);
 		}
