Index: demoscene-toolchain/submodules/fs-uae/src/blitter.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/blitter.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/blitter.cpp
@@ -493,6 +493,8 @@ static void blitter_done (int hpos)
 		write_log (_T("cycles %d, missed %d, total %d\n"),
 			blit_totalcyclecounter, blit_misscyclecounter, blit_totalcyclecounter + blit_misscyclecounter);
 	blitter_dangerous_bpl = 0;
+	log_trace_event((trace_event){.type = 'E', .pid = 0, .tid = 6},
+		"Ucycles", blit_totalcyclecounter + blit_misscyclecounter, NULL);
 }
 
 STATIC_INLINE void chipmem_agnus_wput2 (uaecptr addr, uae_u32 w)
@@ -1633,6 +1635,9 @@ static void do_blitter2(int hpos, int co
 	bltstate = BLT_init;
 	blit_slowdown = 0;
 
+	log_trace_event((trace_event){.type = 'B', .pid = 0, .tid = 6, .name = "blit"},
+		"Ppc", M68K_GETPC, "Ihsize", blt_info.hblitsize, "Ivsize", blt_info.vblitsize, NULL);
+
 	unset_special (SPCFLAG_BLTNASTY);
 	if (dmaen (DMA_BLITPRI))
 		set_special (SPCFLAG_BLTNASTY);
Index: demoscene-toolchain/submodules/fs-uae/src/disk.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/disk.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/disk.cpp
@@ -3446,6 +3446,7 @@ static void disk_dmafinished (void)
 			write_log (_T("%d%s"), floppy[dr].mfmpos, dr < MAX_FLOPPY_DRIVES - 1 ? _T(",") : _T(""));
 		write_log (_T("\n"));
 	}
+	log_trace_event((trace_event){.type = 'E', .pid = 0, .tid = 1}, NULL);
 }
 
 static void fetchnextrevolution (drive *drv)
@@ -4225,6 +4226,9 @@ void DSKLEN (uae_u16 v, int hpos)
 				floppy[dr].cyl * 2 + side, floppy[dr].mfmpos, dma_enable, M68K_GETPC);
 			disk_dma_debugmsg ();
 		}
+		log_trace_event((trace_event){.type = 'B', .pid = 0, .tid = 1,
+		 	.name = (dskdmaen == DSKDMA_WRITE ? "write" : "read")},
+		 	"Idrive", dr, "Itrack", floppy[dr].cyl * 2 + side, NULL);
 	}
 
 	for (dr = 0; dr < MAX_FLOPPY_DRIVES; dr++)
Index: demoscene-toolchain/submodules/fs-uae/src/include/debug.h
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/include/debug.h
+++ demoscene-toolchain/submodules/fs-uae/src/include/debug.h
@@ -235,4 +235,13 @@ STATIC_INLINE void activate_debugger (vo
 
 #endif /* DEBUGGER */
 
+/* Trace Event Format */
+struct trace_event {
+	char type, scope;
+	int pid, tid;
+	const char *name;
+};
+
+extern void log_trace_event(struct trace_event ev, ...);
+
 #endif /* UAE_DEBUG_H */
Index: demoscene-toolchain/submodules/fs-uae/src/include/parallel.h
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/include/parallel.h
+++ demoscene-toolchain/submodules/fs-uae/src/include/parallel.h
@@ -5,6 +5,7 @@
 
 extern int isprinter (void);
 extern void doprinter (uae_u8);
+extern void sendprinter (const char *buf, size_t len);
 extern void flushprinter (void);
 extern void closeprinter (void);
 extern int isprinteropen (void);
Index: demoscene-toolchain/submodules/fs-uae/src/od-fs/parser.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/od-fs/parser.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/od-fs/parser.cpp
@@ -268,6 +268,22 @@ void doprinter (uae_u8 val)
 #endif
 }
 
+void sendprinter (const char *buf, size_t len)
+{
+  if (parallel_mode == PARALLEL_MODE_TCP_PRINTER) {
+    if (parallel_tcp_connected()) {
+      for (size_t sent = 0; sent < len; ) {
+        ssize_t n = uae_socket_write(parallel_tcp, buf + sent, len - sent);
+        if (n <= 0) {
+          parallel_tcp_disconnect ();
+          return;
+        }
+        sent += n;
+      }
+    }
+  }
+}
+
 struct uaeserialdata
 {
 	long hCom;
Index: demoscene-toolchain/submodules/fs-uae/src/uaelib.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/fs-uae/src/uaelib.cpp
+++ demoscene-toolchain/submodules/fs-uae/src/uaelib.cpp
@@ -29,6 +29,7 @@
 #include "filesys.h"
 #include "fsemu-control.h"
 #include "drawing.h"
+#include "parallel.h"
 
 /*
 * Returns UAE Version
@@ -347,6 +348,176 @@ static uae_u32 emulib_DebugDMA(uae_u32 m
   return 1;
 }
 
+static void _vlog_printf(char *str, size_t size, size_t *lenp, const char *format, ...) {
+  va_list ap;
+  va_start(ap, format);
+  (*lenp) += vsnprintf(str + (*lenp), size - (*lenp), format, ap);
+  va_end(ap);
+}
+
+static bool trace_event_enabled = false;
+
+#define GETARG() (get_long(m68k_areg(regs, 7) + (1 + (slot++)) * 4))
+#define vlog_printf(format, ...) _vlog_printf(buf, sizeof(buf), &len, format, ##__VA_ARGS__)
+
+static void vlog_trace_event(struct trace_event ev, va_list ap, int slot) {
+  static unsigned long cycles_begin = 0;
+  static bool first_call = true;
+  static char buf[256];
+
+  if (!trace_event_enabled)
+    return;
+
+  size_t len = 0;
+  buf[0] = '\0';
+
+  if (first_call) {
+    vlog_printf("[\n");
+    cycles_begin = get_cycles();
+    first_call = false;
+  }
+
+  unsigned long clock = (currprefs.ntscmode ? CHIPSET_CLOCK_NTSC : CHIPSET_CLOCK_PAL);
+  double ts_us = (double)(get_cycles() - cycles_begin) * 1e6 / (double)(CYCLE_UNIT * clock);
+
+  vlog_printf("{\"ph\": \"%c\", \"pid\": %d, \"tid\": %d, \"ts\": %.3f", ev.type, ev.pid, ev.tid, ts_us);
+
+  if (ev.type == 'B' || ev.type == 'M' || ev.type == 'C' || ev.type == 'i') {
+    vlog_printf(", \"name\": \"%s\"", ev.name);
+  }
+  if (ev.type == 'i') {
+    vlog_printf(", \"s\": \"%c\"", ev.scope);
+  }
+
+  bool empty = true;
+
+  for (;;) {
+    const char *key;
+
+    if (slot) {
+      uae_u32 addr = GETARG();
+      key = addr ? (char *)get_real_address(addr) : NULL;
+    } else {
+      key = va_arg(ap, char *);
+    }
+
+    if (!key)
+      break;
+
+    if (empty) {
+      vlog_printf(", \"args\": {");
+      empty = false;
+    } else {
+      vlog_printf(", ");
+    }
+
+    switch (*key++) {
+      case 'S': { /* string */
+        const char *s = slot ? (char *)get_real_address(GETARG()) : va_arg(ap, char *);
+        vlog_printf("\"%s\": \"%s\"", key, s);
+        break;
+      }
+      case 'I': { /* signed int */
+        int d = slot ? GETARG() : va_arg(ap, int);
+        vlog_printf("\"%s\": %d", key, d);
+        break;
+      }
+      case 'U': { /* unsigned int */
+        unsigned int d = slot ? GETARG() : va_arg(ap, unsigned int);
+        vlog_printf("\"%s\": %u", key, d);
+        break;
+      }
+      case 'P': { /* amiga pointer */
+        uae_u32 ptr = slot ? GETARG() : va_arg(ap, uae_u32);
+        vlog_printf("\"%s\": \"%p\"", key, (void *)ptr);
+        break;
+      }
+    }
+  }
+
+  if (!empty) {
+    vlog_printf("}");
+  }
+
+  vlog_printf("},\n");
+  sendprinter(buf, len);
+}
+
+#undef GETARG
+
+void log_trace_event(struct trace_event ev, ...) {
+  va_list ap;
+  va_start(ap, ev);
+  vlog_trace_event(ev, ap, 0);
+  va_end(ap);
+}
+
+enum {
+  emulib_TraceEventBegin = 43,
+  emulib_TraceEventEnd = 44,
+  emulib_TraceCounter = 45,
+  emulib_TraceInstant = 46,
+  emulib_TraceMetadata = 47,
+};
+
+static uae_u32 emulib_TraceEvent(uae_u32 arg0, uae_u32 arg1, uae_u32 arg2, uae_u32 arg3) {
+  if (!trace_event_enabled) {
+    trace_event_enabled = true;
+
+    log_trace_event((trace_event){.type = 'M', .pid = 0, .tid = 0, .name = "process_name"},
+                    "Sname", "Hardware", NULL);
+    log_trace_event((trace_event){.type = 'M', .pid = 0, .tid = 6, .name = "thread_name"},
+                    "Sname", "Blitter", NULL);
+    log_trace_event((trace_event){.type = 'M', .pid = 0, .tid = 1, .name = "thread_name"},
+                    "Sname", "Disk", NULL);
+  }
+
+  unsigned int hi = (arg1 >> 8) + 1;
+  unsigned int lo = arg1 & 255;
+
+  trace_event ev = {
+    .pid = hi,
+    .tid = hi * 100 + lo,
+  };
+
+  switch (arg0) {
+    case emulib_TraceEventBegin: {
+      ev.type = 'B';
+      ev.name = (char *)get_real_address(arg2);
+      vlog_trace_event(ev, 0, 3);
+      break;
+    }
+    case emulib_TraceEventEnd: {
+      ev.type = 'E';
+      log_trace_event(ev, 0);
+      break;
+    }
+    case emulib_TraceCounter: {
+      ev.type = 'C';
+      ev.name = (char *)get_real_address(arg2);
+      vlog_trace_event(ev, 0, 3);
+      break;
+    }
+    case emulib_TraceInstant: {
+      ev.type = 'i';
+      ev.name = (char *)get_real_address(arg2);
+      ev.scope = arg3;
+      vlog_trace_event(ev, 0, 4);
+      break;
+    }
+    case emulib_TraceMetadata: {
+      ev.type = 'M';
+      ev.name = (char *)get_real_address(arg2);
+      vlog_trace_event(ev, 0, 3);
+      break;
+    }
+    default:
+      break;
+  }
+
+  return 0;
+}
+
 static int native_dos_op(TrapContext *ctx, uae_u32 mode, uae_u32 p1, uae_u32 p2, uae_u32 p3)
 {
 	TCHAR tmp[MAX_DPATH];
@@ -393,6 +564,7 @@ static uae_u32 uaelib_demux_common(TrapC
  		case 40: return emulib_Log(ARG1);
  		case 41: return emulib_WarpMode(ARG1);
  		case 42: return emulib_DebugDMA(ARG1);
+ 		case 43 ... 47: return emulib_TraceEvent(ARG0, ARG1, ARG2, ARG3);
 
 		case 68: return emulib_Minimize();
 		case 69: return emulib_ExecuteNativeCode();
